src/main/java/taxi/service/DriverServiceImpl.java

```diff
@@ -35,4 +37,13 @@ public Driver update(Driver driver) {
     public boolean delete(Long id) {
         return driverDao.delete(id);
     }
+
+    @Override
+    public Driver findByLogin(String login) throws AuthenticationException {
+        Optional<Driver> byLogin = driverDao.findByLogin(login);
+        if (byLogin.isPresent()) {
+            return byLogin.get();
+        }
+        throw new AuthenticationException("Username or password is incorrect");
```

Comment:
Why AuthenticationException here? And do you know anything about password in this method?